test/uri/uri_path_test.cc

```diff
@@ -9,25 +9,25 @@ TEST(URI_path, https_example_url_no_path) {
 TEST(URI_path, https_example_url_slash) {
   const sourcemeta::jsontoolkit::URI uri{"https://example.com/"};
   EXPECT_TRUE(uri.path().has_value());
-  EXPECT_EQ(uri.path().value(), "/");
+  EXPECT_EQ(uri.path().value(), "");
```

Comment:
@tony-go I think the test at the bottom of this file (for `../foo/bar`) is good after the update, but the rest should have the leading `/`, as that's technically part of the path.

As a consequence, right now you cannot distinguish between i.e. `https://example.com/` and `https://example.com`, as the paths would both be the empty string.

Can you make sure that if the paths are not relative, they indeed start with `/`?

Reply:
But this is overall why I wish there was a really good existing URI test suite we could pull...

Reply:
At least https://github.com/hyperjump-io/uri, which is a compliant URI implementation I trust (from another TSC member of JSON Schema) seems to keep the leading `/`

Reply:
we should see if there is a leading `/` during parsing and storing it.

Reply:
For the test suite, lets do it into a subsequent PR.

Reply:
What do you mean for the test suite?

Reply:
Including the test suit in jsontoolkit project

Reply:
> we should see if there is a leading `/` during parsing and storing it.

i still not resolved it btw. I probably need a few more sessions 😅

Reply:
> Including the test suit in jsontoolkit project

What I meant is that there is no existing URI test suite out there we can use that is good, and that's the challenge in all of this. We have to compare and research a lot more. It'd be great if there was a very solid and very compliant URI parsing suite out there

Reply:
Sorry, I got confused when you said

> I wish there was a really good existing URI

I thought it was locally in the project 🤪